httpcore5/src/test/java/org/apache/hc/core5/ssl/TestSSLContextBuilder.java

```diff
@@ -525,11 +525,11 @@ void testSSLHandshakeClientUnauthenticatedError() throws Exception {
         try (final SSLSocket clientSocket = (SSLSocket) clientSslContext.getSocketFactory().createSocket()) {
             clientSocket.connect(new InetSocketAddress("localhost", localPort), TIMEOUT.toMillisecondsIntBound());
             clientSocket.setSoTimeout(TIMEOUT.toMillisecondsIntBound());
-            Assertions.assertThrows(IOException.class, () -> {
-                clientSocket.startHandshake();
-                final InputStream inputStream = clientSocket.getInputStream();
-                inputStream.read();
-            });
+
```

Comment:
@strangelookingnerd I am not sure this is correct. `#startHanshake` should be inside the asserted lambda. Is there any particular reason for changing the test case other than being able to use the method reference?

Reply:
@strangelookingnerd Otherwise looks good to me

Reply:
The lambda in `Assertions.assertThrows` should only include one call that could throw an exception. So for example:

```java
@Test
void divideByZero() {
  Assertions.assertThrows(DivideByZeroException.class, () -> {
    int a = 4 / 4;
    int b = 4 / 0;
    int c = 4 / 1;
  }
}
```

Here there would be 3 possible method calls where an exception _could_ be thrown. 
My change would make sure that the exception is thrown in `InputStream#read()` and not in `SSLSocket#startHandshake()` - because this is not what is expected.

Reply:
@strangelookingnerd Again, sounds correct in theory. But in this case, different JSSE implementations behave differently even though `#startHandshake` is expected to fail, sometimes it does not until an attempt is made to read from the underlying stream. Please keep this test as it is and will happily merge the rest.

Reply:
Corrected in [960f586](https://github.com/apache/httpcomponents-core/pull/470/commits/960f586b38fb801287f1b203b0e0472fbe36d763)